dotCMS/src/integration-test/java/com/dotcms/rest/BundleResourceTest.java

```
@@ -182,6 +189,56 @@ public void test_generateBundle_success() throws DotDataException, IOException {
+     */
+    @Test
+    public void test_generateBundle_without_SystemHost_and_SystemFolder_references() throws Exception {
+        //Create new bundle
+        final String bundleId = insertPublishingBundle(adminUser.getUserId(),new Date());
+
+        //Create a Filter since it's needed to generate the bundle
+        createFilter();
+
+        //Generate bundle file
+        final Bundle bundle = APILocator.getBundleAPI().getBundleById(bundleId);
+        bundle.setOperation(PushPublisherConfig.Operation.PUBLISH.ordinal());
+        final File bundleFile = APILocator.getBundleAPI().generateTarGzipBundleFile(bundle);
+
+        PublisherAPIImpl.class.cast(APILocator.getPublisherAPI()).getFilterDescriptorMap().clear();
+
+        //Call download endpoint
+        final Response responseResource = bundleResource.downloadBundle(getHttpRequest(),response,bundleId);
+        Assert.assertEquals(Status.OK.getStatusCode(),responseResource.getStatus());
+
+        assertTrue(bundleFile.exists());
+
+        String bundleDir = bundleFile.getAbsolutePath().substring(0, bundleFile.getAbsolutePath().indexOf(".tar.gz"));
+        recursiveFindAndRun(
+                Paths.get(bundleDir),
+                path -> Assert.assertFalse("System Folder".equalsIgnoreCase(path.toFile().getName()) ||
+                        "System Host".equalsIgnoreCase(path.toFile().getName())));
+
+
+    }
+
+    private static void recursiveFindAndRun(final Path path, final Consumer<Path> c) {
+        try (DirectoryStream<Path> newDirectoryStream = Files.newDirectoryStream(path)) {
+            StreamSupport
+                    .stream(newDirectoryStream.spliterator(), false)
+                    .peek(p -> {
+                        c.accept(p);
+                        if (p.toFile().isDirectory()) {
+                            recursiveFindAndRun(p, c);
+                        }
+                    })
+                    .collect(Collectors.toList());
+        } catch (IOException e) {}
```

Codacy found an issue: [Avoid empty catch blocks](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6340082)